### PR TITLE
[WebApp] Add Profile page google sso linking

### PR DESCRIPTION
**Pull request by rners01**
**Description:** Add Profile page google sso linking
**Project:** https://www.notion.so/Frontend-Google-SSO-Account-Connections-38e4c91aaf184133a71187ce594e3994

image
image
image
image
image

### CODE REVIEW

**Review by vitto-moz — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/profile/ProfileStore.tsx [hunk 1]
```diff
@@ -238,6 +256,57 @@ export class ProfileStore {
...
+    const externalAuthProviderLoginAction = urlSearchParams.get('external-login-action')
+
+    this.router.replace('/account/summary')
+
+    if (
+      externalAuthProviderLoginAction === ExternalAuthProviderLoginAction.OneTimeCodeFlow ||
+      externalAuthProviderLoginAction === ExternalAuthProviderLoginAction.WithMessageConfirmation
+    ) {
+      if (externalAuthProviderLoginStatus === ExternalAuthProviderLoginStatus.Success) {
+        return this.store.auth.successfulExternalProviderConnection()
```
Comment (Maks19) [inline packages/web-app/src/modules/profile/ProfileStore.tsx:272]: Why do you need `return`?

File: packages/web-app/src/modules/profile/ProfileStore.tsx [hunk 2]
```diff
@@ -238,6 +256,57 @@ export class ProfileStore {
...
+      }
+
+      if (externalAuthProviderLoginStatus === ExternalAuthProviderLoginStatus.Failed) {
+        return this.store.auth.failedExternalProviderConnection()
```
Comment (Maks19) [inline packages/web-app/src/modules/profile/ProfileStore.tsx:276]: Why do you need `return`?

File: packages/web-app/src/modules/profile/ProfileStore.tsx [hunk 3]
```diff
@@ -238,6 +256,57 @@ export class ProfileStore {
...
+  loadGoogleAccountConnection = flow(function* (this: ProfileStore) {
+    try {
+      this.isLoadConnectedGoogleAccountEmailError = false
+      this.connectExternalAccountProvider()
+
+      const authProviderConnections: AxiosResponse<ExternalAuthProvider[]> = yield this.axios.get(
+        '/api/v2/authentication/external',
+      )
+
+      if (authProviderConnections.data && authProviderConnections.data.length > 0) {
```
Comment (Maks19) [inline packages/web-app/src/modules/profile/ProfileStore.tsx:291]:
```suggestion
      if (authProviderConnections.data?.length > 0) {
```

File: packages/web-app/src/modules/account-views/account-views/components/Account.tsx [hunk 4]
```diff
@@ -74,6 +92,9 @@ interface Props extends WithStyles<typeof styles> {
   isMinecraftUserNameSubmitSuccess: boolean
   payPalId?: string
   loadPayPalId: () => void
+  connectedGoogleAccountEmail: string
```
Comment (Maks19) [inline packages/web-app/src/modules/account-views/account-views/components/Account.tsx:95]:
```suggestion
  connectedGoogleAccountEmail?: string
```

File: packages/web-app/src/modules/account-views/account-views/components/GoogleSignInForm.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+export interface GoogleSignInFormProps extends WithStyles<typeof styles> {
+  isTermsAndConditionsAccepted: boolean
+  isTermsAndConditionsRequired: boolean
+}
+export const _GoogleSignInForm: FunctionComponent<GoogleSignInFormProps> = ({
+  classes,
+  isTermsAndConditionsAccepted,
+  isTermsAndConditionsRequired,
+}) => {
+  const shouldEnableGoogleSignInButton = !isTermsAndConditionsRequired || isTermsAndConditionsAccepted
```
Comment (Maks19) [inline packages/web-app/src/modules/account-views/account-views/components/GoogleSignInForm.tsx:62]: Do we have conditional when `TermsAndConditions` **is not** required and **is not** accepted?

**Review by Maks19 — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/profile/ProfileStore.tsx (hunk 1 above)
Comment (rners01) [reply packages/web-app/src/modules/profile/ProfileStore.tsx:272]: to stop further execution if there is any

Comment (rners01) [reply packages/web-app/src/modules/profile/ProfileStore.tsx:272]: I suppose before that there was some code under but now its not required

File: packages/web-app/src/modules/account-views/account-views/components/Account.tsx [hunk 6]
```diff
@@ -10,36 +10,42 @@ import { Head } from '../../../../components'
...
   },
   headingContainer: {
-    color: theme.darkBlue,
-    paddingBottom: '25px',
+    paddingBottom: 32,
   },
   subheadingContainer: {
-    color: theme.darkBlue,
-    paddingBottom: '7px',
+    paddingBottom: 40,
```
[the remark this answers is not in the document]
Comment (rners01) [reply packages/web-app/src/modules/account-views/account-views/components/Account.tsx:26]: yes, changed to the `number` variant

File: packages/web-app/src/modules/profile/ProfileStore.tsx (hunk 2 above)
Comment (rners01) [reply packages/web-app/src/modules/profile/ProfileStore.tsx]: removed

File: packages/web-app/src/modules/profile/ProfileStore.tsx (hunk 1 above)
Comment (rners01) [reply packages/web-app/src/modules/profile/ProfileStore.tsx]: removed

**Review by vitto-moz — APPROVED** [review]

**Review by Maks19 — DISMISSED** [review]

**Review by Maks19 — APPROVED** [review]

**Pull request merged**